### PR TITLE
added button to share the results

### PR DESCRIPTION
**Pull request by Eliran-dev**
Now users can share the results and view them in links such as https://ironswords.org.il/?name=אבי

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "lucide-react": "^0.284.0",
     "next": "13.5.4",
     "next-themes": "^0.2.1",
-    "react": "^18",
+    "react": "^18.2.0",
```
Comment (vithano) [inline package.json:31]: why?

File: src/components/search.tsx [hunk 2]
```diff
@@ -44,14 +69,23 @@ export function Search({setData, setMessage}: {setData: (data: PersonData[]) =>
...
       placeholder="שם פרטי/משפחה (בעברית)"
       className="md:w-[100px] lg:w-[300px]"
+      defaultValue={searchName}
       onChange={onInputChange}
       isLoading={isLoading}
-      iconSrc={'/search.svg'}
+      iconSrc={"/search.svg"}
     />
+    {isResults &&
+    <Button className="f" onClick={() => onCopy()}>העתק את התוצאות</Button>
```
Comment (vithano) [inline src/components/search.tsx:84]: why do you need to copy the results?

File: src/components/search.tsx [hunk 3]
```diff
@@ -44,14 +69,23 @@ export function Search({setData, setMessage}: {setData: (data: PersonData[]) =>
...
       type="search"
       placeholder="שם פרטי/משפחה (בעברית)"
       className="md:w-[100px] lg:w-[300px]"
+      defaultValue={searchName}
       onChange={onInputChange}
       isLoading={isLoading}
-      iconSrc={'/search.svg'}
+      iconSrc={"/search.svg"}
     />
+    {isResults &&
```
Comment (vithano) [inline src/components/search.tsx:83]: you can just use data.length instead no need for another state

File: src/components/search.tsx [hunk 4]
```diff
@@ -5,12 +5,35 @@ import PersonData from "../app/utils/types";
...
+  const searchParams = useSearchParams();
+
   const inputValueRef = useRef('');
   const [isLoading, setIsLoading] = useState(false);
+  const [searchName, setSearchName] = useState("")
+  const [isResults, setIsResults] = useState(false)
+
+  useEffect(() => {
+    const name = searchParams.get("name")
+    if(name && name.length >= MIN_QUERY_LENGTH)
```
Comment (vithano) [inline src/components/search.tsx:25]: this is already checked in the onInputChange no need to do it twice

**Comment by vithano** [comment]
few comments but overall looks good, thanks!

File: src/components/search.tsx (hunk 3 above)
Comment (Eliran-dev) [reply src/components/search.tsx:83]: Data is a state in the main page, not inside the search component, i think another state will be better than to transfer it as props

**Pull request merged**